### PR TITLE
Python: Update kernel tests

### PR DESCRIPTION
**Pull request by eavanvalkenburg**
### Motivation and Context

Adding a whole bunch of tests to better cover Kernel.py.
Also reviewed and updated the conftest and tweaked the scope, now runs faster!

Skipped efforts for handlers since work is being done there. 

### Contribution Checklist

- [x] The code builds clean without any errors or warnings
- [x] The PR follows the [SK Contribution Guidelines](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [x] All unit tests pass, and I have added new tests where possible
- [x] I didn't break anyone :smile:

### CODE REVIEW

**Comment by markwallace-microsoft** [comment]
<a href="https://github.com/microsoft/semantic-kernel/blob/7c636e0b56f2a53bc046d13db7c37ef2619d5fa6/README.md">Python Test Coverage</a><details><summary>Python Test Coverage Report • </summary><table><tr><th>File</th><th>Stmts</th><th>Miss</th><th>Cover</th><th>Missing</th></tr><tbody><tr><td colspan="5"><b>semantic_kernel</b></td></tr><tr><td>&nbsp; &nbsp;<a href="https://github.com/microsoft/semantic-kernel/blob/7c636e0b56f2a53bc046d13db7c37ef2619d5fa6/python/semantic_kernel/kernel.py">kernel.py</a></td><td>310</td><td>71</td><td>77%</td><td><a href="https://github.com/microsoft/semantic-kernel/blob/7c636e0b56f2a53bc046d13db7c37ef2619d5fa6/python/semantic_kernel/kernel.py#L113">113</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/7c636e0b56f2a53bc046d13db7c37ef2619d5fa6/python/semantic_kernel/kernel.py#L129-L133">129&ndash;133</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/7c636e0b56f2a53bc046d13db7c37ef2619d5fa6/python/semantic_kernel/kernel.py#L167-L179">167&ndash;179</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/7c636e0b56f2a53bc046d13db7c37ef2619d5fa6/python/semantic_kernel/kernel.py#L183-L187">183&ndash;187</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/7c636e0b56f2a53bc046d13db7c37ef2619d5fa6/python/semantic_kernel/kernel.py#L189-L193">189&ndash;193</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/7c636e0b56f2a53bc046d13db7c37ef2619d5fa6/python/semantic_kernel/kernel.py#L195-L199">195&ndash;199</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/7c636e0b56f2a53bc046d13db7c37ef2619d5fa6/python/semantic_kernel/kernel.py#L207-L209">207&ndash;209</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/7c636e0b56f2a53bc046d13db7c37ef2619d5fa6/python/semantic_kernel/kernel.py#L228">228</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/7c636e0b56f2a53bc046d13db7c37ef2619d5fa6/python/semantic_kernel/kernel.py#L234">234</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/7c636e0b56f2a53bc046d13db7c37ef2619d5fa6/python/semantic_kernel/kernel.py#L236-L240">236&ndash;240</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/7c636e0b56f2a53bc046d13db7c37ef2619d5fa6/python/semantic_kernel/kernel.py#L242-L246">242&ndash;246</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/7c636e0b56f2a53bc046d13db7c37ef2619d5fa6/python/semantic_kernel/kernel.py#L248-L253">248&ndash;253</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/7c636e0b56f2a53bc046d13db7c37ef2619d5fa6/python/semantic_kernel/kernel.py#L255">255</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/7c636e0b56f2a53bc046d13db7c37ef2619d5fa6/python/semantic_kernel/kernel.py#L290-L294">290&ndash;294</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/7c636e0b56f2a53bc046d13db7c37ef2619d5fa6/python/semantic_kernel/kernel.py#L311-L316">311&ndash;316</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/7c636e0b56f2a53bc046d13db7c37ef2619d5fa6/python/semantic_kernel/kernel.py#L323">323</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/7c636e0b56f2a53bc046d13db7c37ef2619d5fa6/python/semantic_kernel/kernel.py#L327-L331">327&ndash;331</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/7c636e0b56f2a53bc046d13db7c37ef2619d5fa6/python/semantic_kernel/kernel.py#L344">344</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/7c636e0b56f2a53bc046d13db7c37ef2619d5fa6/python/semantic_kernel/kernel.py#L427-L428">427&ndash;428</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/7c636e0b56f2a53bc046d13db7c37ef2619d5fa6/python/semantic_kernel/kernel.py#L431-L432">431&ndash;432</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/7c636e0b56f2a53bc046d13db7c37ef2619d5fa6/python/semantic_kernel/kernel.py#L472">472</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/7c636e0b56f2a53bc046d13db7c37ef2619d5fa6/python/semantic_kernel/kernel.py#L478">478</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/7c636e0b56f2a53bc046d13db7c37ef2619d5fa6/python/semantic_kernel/kernel.py#L494">494</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/7c636e0b56f2a53bc046d13db7c37ef2619d5fa6/python/semantic_kernel/kernel.py#L512">512</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/7c636e0b56f2a53bc046d13db7c37ef2619d5fa6/python/semantic_kernel/kernel.py#L528">528</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/7c636e0b56f2a53bc046d13db7c37ef2619d5fa6/python/semantic_kernel/kernel.py#L547">547</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/7c636e0b56f2a53bc046d13db7c37ef2619d5fa6/python/semantic_kernel/kernel.py#L559">559</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/7c636e0b56f2a53bc046d13db7c37ef2619d5fa6/python/semantic_kernel/kernel.py#L592-L596">592&ndash;596</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/7c636e0b56f2a53bc046d13db7c37ef2619d5fa6/python/semantic_kernel/kernel.py#L634">634</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/7c636e0b56f2a53bc046d13db7c37ef2619d5fa6/python/semantic_kernel/kernel.py#L719">719</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/7c636e0b56f2a53bc046d13db7c37ef2619d5fa6/python/semantic_kernel/kernel.py#L721-L727">721&ndash;727</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/7c636e0b56f2a53bc046d13db7c37ef2619d5fa6/python/semantic_kernel/kernel.py#L741-L742">741&ndash;742</a></td></tr><tr><td colspan="5"><b>semantic_kernel/functions</b></td></tr><tr><td>&nbsp; &nbsp;<a href="https://github.com/microsoft/semantic-kernel/blob/7c636e0b56f2a53bc046d13db7c37ef2619d5fa6/python/semantic_kernel/functions/kernel_plugin.py">kernel_plugin.py</a></td><td>34</td><td>2</td><td>94%</td><td><a href="https://github.com/microsoft/semantic-kernel/blob/7c636e0b56f2a53bc046d13db7c37ef2619d5fa6/python/semantic_kernel/functions/kernel_plugin.py#L7">7</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/7c636e0b56f2a53bc046d13db7c37ef2619d5fa6/python/semantic_kernel/functions/kernel_plugin.py#L123">123</a></td></tr><tr><td><b>TOTAL</b></td><td><b>4871</b></td><td><b>1040</b></td><td><b>79%</b></td><td>&nbsp;</td></tr></tbody></table></details>

## Python Unit Test Overview
| Tests | Skipped | Failures | Errors | Time |
| ----- | ------- | -------- | -------- | ------------------ |
| 993 | 11 :zzz: | 0 :x: | 0 :fire: | 21.461s :stopwatch: |

**Comment by markwallace-microsoft** [comment]
<a href="https://github.com/microsoft/semantic-kernel/blob/f8dfed245d95e66669af727acf909124d9aba68f/README.md">Python Test Coverage</a><details><summary>Python Test Coverage Report • </summary><table><tr><th>File</th><th>Stmts</th><th>Miss</th><th>Cover</th><th>Missing</th></tr><tbody><tr><td colspan="5"><b>semantic_kernel</b></td></tr><tr><td>&nbsp; &nbsp;<a href="https://github.com/microsoft/semantic-kernel/blob/f8dfed245d95e66669af727acf909124d9aba68f/python/semantic_kernel/kernel.py">kernel.py</a></td><td>310</td><td>71</td><td>77%</td><td><a href="https://github.com/microsoft/semantic-kernel/blob/f8dfed245d95e66669af727acf909124d9aba68f/python/semantic_kernel/kernel.py#L113">113</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/f8dfed245d95e66669af727acf909124d9aba68f/python/semantic_kernel/kernel.py#L129-L133">129&ndash;133</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/f8dfed245d95e66669af727acf909124d9aba68f/python/semantic_kernel/kernel.py#L167-L179">167&ndash;179</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/f8dfed245d95e66669af727acf909124d9aba68f/python/semantic_kernel/kernel.py#L183-L187">183&ndash;187</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/f8dfed245d95e66669af727acf909124d9aba68f/python/semantic_kernel/kernel.py#L189-L193">189&ndash;193</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/f8dfed245d95e66669af727acf909124d9aba68f/python/semantic_kernel/kernel.py#L195-L199">195&ndash;199</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/f8dfed245d95e66669af727acf909124d9aba68f/python/semantic_kernel/kernel.py#L207-L209">207&ndash;209</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/f8dfed245d95e66669af727acf909124d9aba68f/python/semantic_kernel/kernel.py#L228">228</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/f8dfed245d95e66669af727acf909124d9aba68f/python/semantic_kernel/kernel.py#L234">234</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/f8dfed245d95e66669af727acf909124d9aba68f/python/semantic_kernel/kernel.py#L236-L240">236&ndash;240</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/f8dfed245d95e66669af727acf909124d9aba68f/python/semantic_kernel/kernel.py#L242-L246">242&ndash;246</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/f8dfed245d95e66669af727acf909124d9aba68f/python/semantic_kernel/kernel.py#L248-L253">248&ndash;253</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/f8dfed245d95e66669af727acf909124d9aba68f/python/semantic_kernel/kernel.py#L255">255</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/f8dfed245d95e66669af727acf909124d9aba68f/python/semantic_kernel/kernel.py#L290-L294">290&ndash;294</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/f8dfed245d95e66669af727acf909124d9aba68f/python/semantic_kernel/kernel.py#L311-L316">311&ndash;316</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/f8dfed245d95e66669af727acf909124d9aba68f/python/semantic_kernel/kernel.py#L323">323</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/f8dfed245d95e66669af727acf909124d9aba68f/python/semantic_kernel/kernel.py#L327-L331">327&ndash;331</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/f8dfed245d95e66669af727acf909124d9aba68f/python/semantic_kernel/kernel.py#L344">344</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/f8dfed245d95e66669af727acf909124d9aba68f/python/semantic_kernel/kernel.py#L427-L428">427&ndash;428</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/f8dfed245d95e66669af727acf909124d9aba68f/python/semantic_kernel/kernel.py#L431-L432">431&ndash;432</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/f8dfed245d95e66669af727acf909124d9aba68f/python/semantic_kernel/kernel.py#L472">472</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/f8dfed245d95e66669af727acf909124d9aba68f/python/semantic_kernel/kernel.py#L478">478</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/f8dfed245d95e66669af727acf909124d9aba68f/python/semantic_kernel/kernel.py#L494">494</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/f8dfed245d95e66669af727acf909124d9aba68f/python/semantic_kernel/kernel.py#L512">512</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/f8dfed245d95e66669af727acf909124d9aba68f/python/semantic_kernel/kernel.py#L528">528</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/f8dfed245d95e66669af727acf909124d9aba68f/python/semantic_kernel/kernel.py#L547">547</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/f8dfed245d95e66669af727acf909124d9aba68f/python/semantic_kernel/kernel.py#L559">559</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/f8dfed245d95e66669af727acf909124d9aba68f/python/semantic_kernel/kernel.py#L592-L596">592&ndash;596</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/f8dfed245d95e66669af727acf909124d9aba68f/python/semantic_kernel/kernel.py#L634">634</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/f8dfed245d95e66669af727acf909124d9aba68f/python/semantic_kernel/kernel.py#L719">719</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/f8dfed245d95e66669af727acf909124d9aba68f/python/semantic_kernel/kernel.py#L721-L727">721&ndash;727</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/f8dfed245d95e66669af727acf909124d9aba68f/python/semantic_kernel/kernel.py#L741-L742">741&ndash;742</a></td></tr><tr><td colspan="5"><b>semantic_kernel/functions</b></td></tr><tr><td>&nbsp; &nbsp;<a href="https://github.com/microsoft/semantic-kernel/blob/f8dfed245d95e66669af727acf909124d9aba68f/python/semantic_kernel/functions/kernel_plugin.py">kernel_plugin.py</a></td><td>34</td><td>2</td><td>94%</td><td><a href="https://github.com/microsoft/semantic-kernel/blob/f8dfed245d95e66669af727acf909124d9aba68f/python/semantic_kernel/functions/kernel_plugin.py#L7">7</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/f8dfed245d95e66669af727acf909124d9aba68f/python/semantic_kernel/functions/kernel_plugin.py#L123">123</a></td></tr><tr><td><b>TOTAL</b></td><td><b>4871</b></td><td><b>1040</b></td><td><b>79%</b></td><td>&nbsp;</td></tr></tbody></table></details>

## Python Unit Test Overview
| Tests | Skipped | Failures | Errors | Time |
| ----- | ------- | -------- | -------- | ------------------ |
| 993 | 11 :zzz: | 0 :x: | 0 :fire: | 24.939s :stopwatch: |

**Comment by markwallace-microsoft** [comment]
<a href="https://github.com/microsoft/semantic-kernel/blob/144871268f2b2103ba860d93fff0b6f092f56c84/README.md">Python Test Coverage</a><details><summary>Python Test Coverage Report • </summary><table><tr><th>File</th><th>Stmts</th><th>Miss</th><th>Cover</th><th>Missing</th></tr><tbody><tr><td colspan="5"><b>semantic_kernel</b></td></tr><tr><td>&nbsp; &nbsp;<a href="https://github.com/microsoft/semantic-kernel/blob/144871268f2b2103ba860d93fff0b6f092f56c84/python/semantic_kernel/kernel.py">kernel.py</a></td><td>310</td><td>72</td><td>77%</td><td><a href="https://github.com/microsoft/semantic-kernel/blob/144871268f2b2103ba860d93fff0b6f092f56c84/python/semantic_kernel/kernel.py#L113">113</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/144871268f2b2103ba860d93fff0b6f092f56c84/python/semantic_kernel/kernel.py#L129-L133">129&ndash;133</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/144871268f2b2103ba860d93fff0b6f092f56c84/python/semantic_kernel/kernel.py#L161">161</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/144871268f2b2103ba860d93fff0b6f092f56c84/python/semantic_kernel/kernel.py#L167-L179">167&ndash;179</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/144871268f2b2103ba860d93fff0b6f092f56c84/python/semantic_kernel/kernel.py#L183-L187">183&ndash;187</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/144871268f2b2103ba860d93fff0b6f092f56c84/python/semantic_kernel/kernel.py#L189-L193">189&ndash;193</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/144871268f2b2103ba860d93fff0b6f092f56c84/python/semantic_kernel/kernel.py#L195-L199">195&ndash;199</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/144871268f2b2103ba860d93fff0b6f092f56c84/python/semantic_kernel/kernel.py#L207-L209">207&ndash;209</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/144871268f2b2103ba860d93fff0b6f092f56c84/python/semantic_kernel/kernel.py#L228">228</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/144871268f2b2103ba860d93fff0b6f092f56c84/python/semantic_kernel/kernel.py#L234">234</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/144871268f2b2103ba860d93fff0b6f092f56c84/python/semantic_kernel/kernel.py#L236-L240">236&ndash;240</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/144871268f2b2103ba860d93fff0b6f092f56c84/python/semantic_kernel/kernel.py#L242-L246">242&ndash;246</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/144871268f2b2103ba860d93fff0b6f092f56c84/python/semantic_kernel/kernel.py#L248-L253">248&ndash;253</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/144871268f2b2103ba860d93fff0b6f092f56c84/python/semantic_kernel/kernel.py#L255">255</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/144871268f2b2103ba860d93fff0b6f092f56c84/python/semantic_kernel/kernel.py#L290-L294">290&ndash;294</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/144871268f2b2103ba860d93fff0b6f092f56c84/python/semantic_kernel/kernel.py#L311-L316">311&ndash;316</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/144871268f2b2103ba860d93fff0b6f092f56c84/python/semantic_kernel/kernel.py#L323">323</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/144871268f2b2103ba860d93fff0b6f092f56c84/python/semantic_kernel/kernel.py#L327-L331">327&ndash;331</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/144871268f2b2103ba860d93fff0b6f092f56c84/python/semantic_kernel/kernel.py#L344">344</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/144871268f2b2103ba860d93fff0b6f092f56c84/python/semantic_kernel/kernel.py#L427-L428">427&ndash;428</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/144871268f2b2103ba860d93fff0b6f092f56c84/python/semantic_kernel/kernel.py#L431-L432">431&ndash;432</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/144871268f2b2103ba860d93fff0b6f092f56c84/python/semantic_kernel/kernel.py#L472">472</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/144871268f2b2103ba860d93fff0b6f092f56c84/python/semantic_kernel/kernel.py#L478">478</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/144871268f2b2103ba860d93fff0b6f092f56c84/python/semantic_kernel/kernel.py#L494">494</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/144871268f2b2103ba860d93fff0b6f092f56c84/python/semantic_kernel/kernel.py#L512">512</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/144871268f2b2103ba860d93fff0b6f092f56c84/python/semantic_kernel/kernel.py#L528">528</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/144871268f2b2103ba860d93fff0b6f092f56c84/python/semantic_kernel/kernel.py#L547">547</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/144871268f2b2103ba860d93fff0b6f092f56c84/python/semantic_kernel/kernel.py#L559">559</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/144871268f2b2103ba860d93fff0b6f092f56c84/python/semantic_kernel/kernel.py#L592-L596">592&ndash;596</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/144871268f2b2103ba860d93fff0b6f092f56c84/python/semantic_kernel/kernel.py#L634">634</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/144871268f2b2103ba860d93fff0b6f092f56c84/python/semantic_kernel/kernel.py#L719">719</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/144871268f2b2103ba860d93fff0b6f092f56c84/python/semantic_kernel/kernel.py#L721-L727">721&ndash;727</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/144871268f2b2103ba860d93fff0b6f092f56c84/python/semantic_kernel/kernel.py#L741-L742">741&ndash;742</a></td></tr><tr><td colspan="5"><b>semantic_kernel/functions</b></td></tr><tr><td>&nbsp; &nbsp;<a href="https://github.com/microsoft/semantic-kernel/blob/144871268f2b2103ba860d93fff0b6f092f56c84/python/semantic_kernel/functions/kernel_plugin.py">kernel_plugin.py</a></td><td>34</td><td>2</td><td>94%</td><td><a href="https://github.com/microsoft/semantic-kernel/blob/144871268f2b2103ba860d93fff0b6f092f56c84/python/semantic_kernel/functions/kernel_plugin.py#L7">7</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/144871268f2b2103ba860d93fff0b6f092f56c84/python/semantic_kernel/functions/kernel_plugin.py#L123">123</a></td></tr><tr><td><b>TOTAL</b></td><td><b>4850</b></td><td><b>1020</b></td><td><b>79%</b></td><td>&nbsp;</td></tr></tbody></table></details>

## Python Unit Test Overview
| Tests | Skipped | Failures | Errors | Time |
| ----- | ------- | -------- | -------- | ------------------ |
| 990 | 11 :zzz: | 0 :x: | 0 :fire: | 22.632s :stopwatch: |

**Comment by markwallace-microsoft** [comment]
<a href="https://github.com/microsoft/semantic-kernel/blob/1927991da1772d4e548535ee16d9b21cec62466f/README.md">Python Test Coverage</a><details><summary>Python Test Coverage Report • </summary><table><tr><th>File</th><th>Stmts</th><th>Miss</th><th>Cover</th><th>Missing</th></tr><tbody><tr><td colspan="5"><b>semantic_kernel</b></td></tr><tr><td>&nbsp; &nbsp;<a href="https://github.com/microsoft/semantic-kernel/blob/1927991da1772d4e548535ee16d9b21cec62466f/python/semantic_kernel/kernel.py">kernel.py</a></td><td>310</td><td>72</td><td>77%</td><td><a href="https://github.com/microsoft/semantic-kernel/blob/1927991da1772d4e548535ee16d9b21cec62466f/python/semantic_kernel/kernel.py#L113">113</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/1927991da1772d4e548535ee16d9b21cec62466f/python/semantic_kernel/kernel.py#L129-L133">129&ndash;133</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/1927991da1772d4e548535ee16d9b21cec62466f/python/semantic_kernel/kernel.py#L161">161</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/1927991da1772d4e548535ee16d9b21cec62466f/python/semantic_kernel/kernel.py#L167-L179">167&ndash;179</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/1927991da1772d4e548535ee16d9b21cec62466f/python/semantic_kernel/kernel.py#L183-L187">183&ndash;187</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/1927991da1772d4e548535ee16d9b21cec62466f/python/semantic_kernel/kernel.py#L189-L193">189&ndash;193</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/1927991da1772d4e548535ee16d9b21cec62466f/python/semantic_kernel/kernel.py#L195-L199">195&ndash;199</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/1927991da1772d4e548535ee16d9b21cec62466f/python/semantic_kernel/kernel.py#L207-L209">207&ndash;209</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/1927991da1772d4e548535ee16d9b21cec62466f/python/semantic_kernel/kernel.py#L228">228</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/1927991da1772d4e548535ee16d9b21cec62466f/python/semantic_kernel/kernel.py#L234">234</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/1927991da1772d4e548535ee16d9b21cec62466f/python/semantic_kernel/kernel.py#L236-L240">236&ndash;240</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/1927991da1772d4e548535ee16d9b21cec62466f/python/semantic_kernel/kernel.py#L242-L246">242&ndash;246</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/1927991da1772d4e548535ee16d9b21cec62466f/python/semantic_kernel/kernel.py#L248-L253">248&ndash;253</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/1927991da1772d4e548535ee16d9b21cec62466f/python/semantic_kernel/kernel.py#L255">255</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/1927991da1772d4e548535ee16d9b21cec62466f/python/semantic_kernel/kernel.py#L290-L294">290&ndash;294</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/1927991da1772d4e548535ee16d9b21cec62466f/python/semantic_kernel/kernel.py#L311-L316">311&ndash;316</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/1927991da1772d4e548535ee16d9b21cec62466f/python/semantic_kernel/kernel.py#L323">323</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/1927991da1772d4e548535ee16d9b21cec62466f/python/semantic_kernel/kernel.py#L327-L331">327&ndash;331</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/1927991da1772d4e548535ee16d9b21cec62466f/python/semantic_kernel/kernel.py#L344">344</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/1927991da1772d4e548535ee16d9b21cec62466f/python/semantic_kernel/kernel.py#L427-L428">427&ndash;428</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/1927991da1772d4e548535ee16d9b21cec62466f/python/semantic_kernel/kernel.py#L431-L432">431&ndash;432</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/1927991da1772d4e548535ee16d9b21cec62466f/python/semantic_kernel/kernel.py#L472">472</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/1927991da1772d4e548535ee16d9b21cec62466f/python/semantic_kernel/kernel.py#L478">478</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/1927991da1772d4e548535ee16d9b21cec62466f/python/semantic_kernel/kernel.py#L494">494</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/1927991da1772d4e548535ee16d9b21cec62466f/python/semantic_kernel/kernel.py#L512">512</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/1927991da1772d4e548535ee16d9b21cec62466f/python/semantic_kernel/kernel.py#L528">528</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/1927991da1772d4e548535ee16d9b21cec62466f/python/semantic_kernel/kernel.py#L547">547</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/1927991da1772d4e548535ee16d9b21cec62466f/python/semantic_kernel/kernel.py#L559">559</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/1927991da1772d4e548535ee16d9b21cec62466f/python/semantic_kernel/kernel.py#L592-L596">592&ndash;596</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/1927991da1772d4e548535ee16d9b21cec62466f/python/semantic_kernel/kernel.py#L634">634</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/1927991da1772d4e548535ee16d9b21cec62466f/python/semantic_kernel/kernel.py#L719">719</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/1927991da1772d4e548535ee16d9b21cec62466f/python/semantic_kernel/kernel.py#L721-L727">721&ndash;727</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/1927991da1772d4e548535ee16d9b21cec62466f/python/semantic_kernel/kernel.py#L741-L742">741&ndash;742</a></td></tr><tr><td colspan="5"><b>semantic_kernel/functions</b></td></tr><tr><td>&nbsp; &nbsp;<a href="https://github.com/microsoft/semantic-kernel/blob/1927991da1772d4e548535ee16d9b21cec62466f/python/semantic_kernel/functions/kernel_plugin.py">kernel_plugin.py</a></td><td>34</td><td>2</td><td>94%</td><td><a href="https://github.com/microsoft/semantic-kernel/blob/1927991da1772d4e548535ee16d9b21cec62466f/python/semantic_kernel/functions/kernel_plugin.py#L7">7</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/1927991da1772d4e548535ee16d9b21cec62466f/python/semantic_kernel/functions/kernel_plugin.py#L123">123</a></td></tr><tr><td><b>TOTAL</b></td><td><b>4850</b></td><td><b>1020</b></td><td><b>79%</b></td><td>&nbsp;</td></tr></tbody></table></details>

## Python Unit Test Overview
| Tests | Skipped | Failures | Errors | Time |
| ----- | ------- | -------- | -------- | ------------------ |
| 990 | 11 :zzz: | 0 :x: | 0 :fire: | 18.582s :stopwatch: |

File: python/tests/unit/contents/test_chat_history.py [hunk 1]
```diff
@@ -15,6 +15,11 @@
 from semantic_kernel.prompt_template.prompt_template_config import PromptTemplateConfig
 
 
+@pytest.fixture(scope="function")
+def chat_history():
```
Comment (moonbox3) [inline python/tests/unit/contents/test_chat_history.py:19]: I remember doing this before, and you asked me to remove it. :)

**Review by moonbox3 — APPROVED** [review]

File: python/tests/unit/contents/test_chat_history.py (hunk 1 above)
Comment (eavanvalkenburg) [reply python/tests/unit/contents/test_chat_history.py:19]: oops, my bad 😨

**Review by juliomenendez — APPROVED** [review]

**Pull request merged**